cpp/src/memory/pinned_memory_resource.cu

```diff
@@ -43,7 +44,9 @@ cuda::experimental::memory_pool_properties get_memory_pool_properties() {
 
 struct PinnedMemoryResource::PinnedMemoryResourceImpl {
     PinnedMemoryResourceImpl(int numa_id)
-        : pool{numa_id, get_memory_pool_properties()}, resource{pool} {}
+        : pool{numa_id, get_memory_pool_properties()}, resource{pool} {
+        pool.enable_access_from(rmm::get_current_cuda_device().value());
```

Comment:
Looks like CCCL already makes the pool accessible [from all device](https://github.com/NVIDIA/cccl/blob/f5723522383553b70ccd7eed9a7495cccbc7c9c8/libcudacxx/include/cuda/__memory_pool/pinned_memory_pool.h#L153) ?

Reply:
>  Maybe this is redundant then .

And misleading because the pool is already accessible from all GPUs not just  `rmm::get_current_cuda_device()`.